internal/api/controllers/private/runsCreateHelpers.go

```diff
@@ -0,0 +1,281 @@
+package private
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"net/http"
+
+	"playbook-dispatcher/internal/api/connectors"
+	"playbook-dispatcher/internal/api/controllers/public"
+	"playbook-dispatcher/internal/api/instrumentation"
+	"playbook-dispatcher/internal/api/middleware"
+	dbModel "playbook-dispatcher/internal/common/model/db"
+
+	"github.com/google/uuid"
+	"github.com/spf13/viper"
+	"gorm.io/gorm"
+)
+
```

Comment:
This file is for sure a step in the right direction - i.e. defining the main code path only once. One nitpick would be that since most of the core logic of the API module is now located here it may deserve a more prominent name than "helpers". Related to that is that instead of a set of stateless functions, still orchestrated from controllers, it may be better to turn this into a stateful component on its own, that orchestrates all the actions involved in dispatching of request. A component that orchestrates building of the payload, sending it to cloud connector and persisting the records. That would allow the responsibilities of controllers to be reduced to dealing with HTTP requests/responses and other hand maximize code reuse in the orchestrating component.

Reply:
I renamed this file to `runsCreateActions`. Lmk, if you can think of any more fitting name for this.
I am also looking into how we can make this a stateful component without running into cyclic dependency issues. At this current state, quite a few variables/types are shared between this file and endpoint functions.